app/services/firebase/auth.ts

```diff
@@ -0,0 +1,41 @@
+import auth from '@react-native-firebase/auth';
+import _ from 'lodash';
+
+import { parseFairbaseError } from '../../utils';
+import { IFirebaseAuthResponse } from './types';
+
+export const registerUser = async (email: string, password: string) => {
+  const registerUserResponse: IFirebaseAuthResponse = {
+    user: null,
+    error: null,
+  };
+  await auth()
+    .createUserWithEmailAndPassword(email, password)
+    .then(result => {
+      registerUserResponse.user = _.cloneDeep(result.user);
+    })
+    .catch(error => {
+      registerUserResponse.error = parseFairbaseError(error.code);
+    });
+  return registerUserResponse;
```

Comment:
Seems not properly.
If we use async function -- we shouldn't use `.then` / `.catch`
```
try {
const user = await auth().createUserWithEmailAndPassword(email, password);
return user;
} catch (error: FirebaseErrorType) {
return error;
}
```

In the return type we should specify `IUser | IFirebaseError`